components/fxa-client/src/scoped_keys.rs

```diff
@@ -88,6 +103,66 @@ impl ScopedKeysFlow {
         .to_string())
     }
 
+    fn get_peer_public_from_jwk(jwk: &str) -> Result<Vec<u8>> {
+        let jwk_value: serde_json::Value = serde_json::from_str(jwk)?;
+        let x = jwk_value
+            .as_object()
+            .ok_or_else(|| ErrorKind::ParseJsonError("jwk json is not an object"))?
+            .get("x")
+            .ok_or_else(|| ErrorKind::ParseJsonError("Could not find x in jwk json"))?
+            .as_str()
+            .ok_or_else(|| ErrorKind::ParseJsonError("x value is not a string in jwk json"))?;
+        let y = jwk_value
+            .as_object()
+            .ok_or_else(|| ErrorKind::ParseJsonError("jwk json is not an object"))?
+            .get("y")
+            .ok_or_else(|| ErrorKind::ParseJsonError("Could not find y in jwk json"))?
+            .as_str()
+            .ok_or_else(|| ErrorKind::ParseJsonError("y value is not a string in jwk json"))?;
```

Comment:
Can we replace all of this with `let jwk: Jwk = serde_json::from_str(jwk)`?

Reply:
Yes we can 🤦 I added the Jwk struct after I had written this stuff. Thanks @eoger